aeneas/src/jvm/JvmRep.v3

```diff
@@ -56,7 +56,8 @@ class JvmTypeReps(jprog: JvmProgram) {
 	}
 	private def makeJvmClass(superRep: JvmType, utype: Type) -> JvmType {
 		var ic = jprog.prog.ir.getIrClass(utype);
-		var className = jprog.mangleTypeName("V3C_", utype);
+		var base_name = if (ic == null, "V3C_", StringBuilder.new().put1("V3C_%d", ic.uid).toString());
```

Comment:
Names are useful when crashing on the JVM. I think this would be better if `mangleTypeName` just appends the UID if the class was private.

Reply:
It seems like a type's `isPrivate` info is only accessible at the Verifier, and takes quite a lot of code changes to be made available in the backend, so I think it'll be safer to have `mangleTypeName` append the uid regardless of `isPrivate`

Reply:
You should be able to get the privacy like this:

```
var isPrivate = if(ClassType.?(utype), ClassType.!(utype).classDecl.isPrivate);
```